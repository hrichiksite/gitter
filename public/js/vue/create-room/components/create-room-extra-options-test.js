const mount = require('../../__test__/vuex-mount');
const { default: CreateRoomExtraOptions } = require('./create-room-extra-options.vue');
const { types } = require('../store');

describe('CreateRoomExtraOptions', () => {
  it('matches snapshot', () => {
    const { wrapper } = mount(CreateRoomExtraOptions);
    expect(wrapper.element).toMatchSnapshot();
  });

  describe('onlyGithubUsers', () => {
    it('visible when associated with public GitHub repo', () => {
      const { wrapper } = mount(CreateRoomExtraOptions, {}, store => {
        store.state.createRoom.selectedRepoId = 123;
        store.state.createRoom.adminRepoMap = {
          123: {
            type: 'GH_REPO'
          }
        };
      });
      expect(wrapper.element).toMatchSnapshot();
    });

    it('not visible when no repo association', () => {
      const { wrapper } = mount(CreateRoomExtraOptions, {}, store => {
        store.state.createRoom.selectedRepoId = null;
      });
      expect(wrapper.element).toMatchSnapshot();
    });

    it('clicking onlyGitHubUsers checkbox fires store mutation', () => {
      const { wrapper, stubbedMutations } = mount(CreateRoomExtraOptions, {}, store => {
        store.state.createRoom.selectedRepoId = 123;
        store.state.createRoom.adminRepoMap = {
          123: {
            type: 'GH_REPO'
          }
        };
      });

      wrapper.find({ ref: 'onlyGitHubUsersCheckbox' }).trigger('click');

      expect(stubbedMutations.createRoom[types.SET_ONLY_GITHUB_USERS]).toHaveBeenCalledWith(
        expect.anything(),
        true
      );
    });
  });

  describe('allowGroupAdmins', () => {
    it('visible when private room with no repo association', () => {
      const { wrapper } = mount(CreateRoomExtraOptions, {}, store => {
        store.state.createRoom.roomSecurity = 'PRIVATE';
        store.state.createRoom.selectedRepoId = null;
      });
      expect(wrapper.element).toMatchSnapshot();
    });

    it('not visible when public room', () => {
      const { wrapper } = mount(CreateRoomExtraOptions, {}, store => {
        store.state.createRoom.roomSecurity = 'PUBLIC';
      });
      expect(wrapper.element).toMatchSnapshot();
    });

    it('not visible when associated with GitHub repo', () => {
      const { wrapper } = mount(CreateRoomExtraOptions, {}, store => {
        store.state.createRoom.selectedRepoId = 123;
        store.state.createRoom.adminRepoMap = {
          123: {
            type: 'GH_REPO'
          }
        };
      });
      expect(wrapper.element).toMatchSnapshot();
    });

    it('clicking allowGroupAdminsCheckbox checkbox fires store mutation', () => {
      const { wrapper, stubbedMutations } = mount(CreateRoomExtraOptions, {}, store => {
        store.state.createRoom.roomSecurity = 'PRIVATE';
        store.state.createRoom.selectedRepoId = null;
      });

      wrapper.find({ ref: 'allowGroupAdminsCheckbox' }).trigger('click');

      expect(stubbedMutations.createRoom[types.SET_ALLOW_GROUP_ADMINS]).toHaveBeenCalledWith(
        expect.anything(),
        true
      );
    });
  });

  describe('allowBadger', () => {
    it('visible when associated with public GitHub repo', () => {
      const { wrapper } = mount(CreateRoomExtraOptions, {}, store => {
        store.state.createRoom.selectedRepoId = 123;
        store.state.createRoom.adminRepoMap = {
          123: {
            type: 'GH_REPO'
          }
        };
      });
      expect(wrapper.element).toMatchSnapshot();
    });

    it('not visible no repo association', () => {
      const { wrapper } = mount(CreateRoomExtraOptions, {}, store => {
        store.state.createRoom.selectedRepoId = null;
      });
      expect(wrapper.element).toMatchSnapshot();
    });

    it('not visible when repo is private', () => {
      const { wrapper } = mount(CreateRoomExtraOptions, {}, store => {
        store.state.createRoom.selectedRepoId = 123;
        store.state.createRoom.adminRepoMap = {
          123: {
            type: 'GH_REPO',
            private: true
          }
        };
      });
      expect(wrapper.element).toMatchSnapshot();
    });

    it('clicking onlyGitHubUsers checkbox fires store mutation', () => {
      const { wrapper, stubbedMutations } = mount(CreateRoomExtraOptions, {}, store => {
        store.state.createRoom.allowBadger = false;
        store.state.createRoom.selectedRepoId = 123;
        store.state.createRoom.adminRepoMap = {
          123: {
            type: 'GH_REPO'
          }
        };
      });

      wrapper.find({ ref: 'allowBadgerCheckbox' }).trigger('click');

      expect(stubbedMutations.createRoom[types.SET_ALLOW_BADGER]).toHaveBeenCalledWith(
        expect.anything(),
        true
      );
    });
  });
});
