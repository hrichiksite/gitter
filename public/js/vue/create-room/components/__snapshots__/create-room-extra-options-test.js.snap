// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CreateRoomExtraOptions allowBadger not visible no repo association 1`] = `
<section
  class="create-room-detail-section"
>
  <ul
    class="option-list"
  >
    <!---->
     
    <!---->
     
    <!---->
  </ul>
</section>
`;

exports[`CreateRoomExtraOptions allowBadger not visible when repo is private 1`] = `
<section
  class="create-room-detail-section"
>
  <ul
    class="option-list"
  >
    <li
      class="option-item"
    >
      <input
        class="option-item-input"
        id="create-room-only-github-users"
        type="checkbox"
      />
       
      <label
        class="option-item-label"
        for="create-room-only-github-users"
      >
        
        Only GitHub users can join this room
      
      </label>
    </li>
     
    <!---->
     
    <!---->
  </ul>
</section>
`;

exports[`CreateRoomExtraOptions allowBadger visible when associated with public GitHub repo 1`] = `
<section
  class="create-room-detail-section"
>
  <ul
    class="option-list"
  >
    <li
      class="option-item"
    >
      <input
        class="option-item-input"
        id="create-room-only-github-users"
        type="checkbox"
      />
       
      <label
        class="option-item-label"
        for="create-room-only-github-users"
      >
        
        Only GitHub users can join this room
      
      </label>
    </li>
     
    <!---->
     
    <li
      class="option-item"
    >
      <input
        class="option-item-input"
        id="create-room-allow-badger"
        type="checkbox"
      />
       
      <label
        class="option-item-label"
        for="create-room-allow-badger"
      >
        
        Send PR to add the Gitter badge to your README
      
      </label>
    </li>
  </ul>
</section>
`;

exports[`CreateRoomExtraOptions allowGroupAdmins not visible when associated with GitHub repo 1`] = `
<section
  class="create-room-detail-section"
>
  <ul
    class="option-list"
  >
    <li
      class="option-item"
    >
      <input
        class="option-item-input"
        id="create-room-only-github-users"
        type="checkbox"
      />
       
      <label
        class="option-item-label"
        for="create-room-only-github-users"
      >
        
        Only GitHub users can join this room
      
      </label>
    </li>
     
    <!---->
     
    <li
      class="option-item"
    >
      <input
        class="option-item-input"
        id="create-room-allow-badger"
        type="checkbox"
      />
       
      <label
        class="option-item-label"
        for="create-room-allow-badger"
      >
        
        Send PR to add the Gitter badge to your README
      
      </label>
    </li>
  </ul>
</section>
`;

exports[`CreateRoomExtraOptions allowGroupAdmins not visible when public room 1`] = `
<section
  class="create-room-detail-section"
>
  <ul
    class="option-list"
  >
    <!---->
     
    <!---->
     
    <!---->
  </ul>
</section>
`;

exports[`CreateRoomExtraOptions allowGroupAdmins visible when private room with no repo association 1`] = `
<section
  class="create-room-detail-section"
>
  <ul
    class="option-list"
  >
    <!---->
     
    <li
      class="option-item"
    >
      <input
        class="option-item-input"
        id="create-room-allow-group-admins"
        type="checkbox"
      />
       
      <label
        class="option-item-label"
        for="create-room-allow-group-admins"
      >
        
        Inherit permissions from the community and allow other
        community admins to join this room
      
      </label>
    </li>
     
    <!---->
  </ul>
</section>
`;

exports[`CreateRoomExtraOptions matches snapshot 1`] = `
<section
  class="create-room-detail-section"
>
  <ul
    class="option-list"
  >
    <!---->
     
    <!---->
     
    <!---->
  </ul>
</section>
`;

exports[`CreateRoomExtraOptions onlyGithubUsers not visible when no repo association 1`] = `
<section
  class="create-room-detail-section"
>
  <ul
    class="option-list"
  >
    <!---->
     
    <!---->
     
    <!---->
  </ul>
</section>
`;

exports[`CreateRoomExtraOptions onlyGithubUsers visible when associated with public GitHub repo 1`] = `
<section
  class="create-room-detail-section"
>
  <ul
    class="option-list"
  >
    <li
      class="option-item"
    >
      <input
        class="option-item-input"
        id="create-room-only-github-users"
        type="checkbox"
      />
       
      <label
        class="option-item-label"
        for="create-room-only-github-users"
      >
        
        Only GitHub users can join this room
      
      </label>
    </li>
     
    <!---->
     
    <li
      class="option-item"
    >
      <input
        class="option-item-input"
        id="create-room-allow-badger"
        type="checkbox"
      />
       
      <label
        class="option-item-label"
        for="create-room-allow-badger"
      >
        
        Send PR to add the Gitter badge to your README
      
      </label>
    </li>
  </ul>
</section>
`;
